Release Notes file for 4.15 (#67532)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

|Supported
|Supported

|File Integrity Operator
|Supported
|Supported

|HyperShift Operator
|Technology Preview
|Technology Preview

|Local Storage Operator
|Supported
|Supported

|MetalLB Operator
|Supported
|Supported

|NFD Operator
|Supported
|Supported

|NMState Operator
|Supported
|Supported

|OpenShift Elasticsearch Operator
|Supported
|Supported

|Service Binding Operator
|Supported
|Supported

|Vertical Pod Autoscaler Operator
|Supported
|Supported
|====

.Multus CNI plugins
[cols="2,1,1",options="header"]
|====
|Feature |{ibmpowerRegProductName} |{ibmzRegProductName} and {linuxoneProductName}

|Bridge
|Supported
|Supported

|Host-device
|Supported
|Supported

|IPAM
|Supported
|Supported

|IPVLAN
|Supported
|Supported
|====

.CSI Volumes
[cols="2,1,1",options="header"]
|====
|Feature |{ibmpowerRegProductName} |{ibmzRegProductName} and {linuxoneProductName}

|Cloning
|Supported
|Supported

|Expansion
|Supported
|Supported

|Snapshot
|Supported
|Supported
|====

[id="ocp-4-15-auth"]
=== Authentication and authorization

[id="ocp-4-15-networking"]
=== Networking

[id="ocp-4-15-registry"]
=== Registry

[id="ocp-4-15-storage"]
=== Storage

[id="ocp-4-15-oci"]

> === Oracle&#174; Cloud Infrastructure

Reviewer: This won't work on the portal, replace with (R)
```suggestion
=== Oracle(R) Cloud Infrastructure
```

[id="ocp-4-15-olm"]
=== Operator lifecycle

[id="ocp-4-15-osdk"]
=== Operator development

[id="ocp-4-15-builds"]
=== Builds

[id="ocp-4-15-machine-config-operator"]
=== Machine Config Operator

[id="ocp-4-15-machine-api"]
=== Machine API

[id="ocp-4-15-nodes"]
=== Nodes

[id="ocp-4-15-monitoring"]
=== Monitoring

> The monitoring stack for this release includes the following new and modified features:

Reviewer: Not sure why this one has an intro :thinking:

[id="ocp-4-15-network-observability-1-5"]
=== Network Observability Operator
The Network Observability Operator releases updates independently from the {product-title} minor version release stream. Updates are available through a single, rolling stream which is supported on all currently supported versions of {product-title} 4. Information regarding new features, enhancements, and bug fixes for the Network Observability Operator is found in the xref:../network_observability/network-observability-operator-release-notes.adoc[Network Observability release notes].

[id="ocp-4-15-scalability-and-performance"]
=== Scalability and performance

[id="ocp-4-15-hcp"]

> === Hosted control planes (Technology Preview)

Reviewer 2: @JoeAldinger That can go away for 4.15. Thanks!

.{rh-openstack} Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.13 |4.14 |4.15

|External load balancers with installer-provisioned infrastructure
|Technology Preview
|General Availability
|General Availability

|Dual-stack networking with installer-provisioned infrastructure
|Not Available
|Technology Preview
|Technology Preview

|====

[discrete]
=== Architecture Technology Preview features

.Architecture Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.13 |4.14 |4.15

|Hosted control planes for {product-title} on Amazon Web Services (AWS)
|Technology Preview
|Technology Preview
|Technology Preview
// Needs to move to GA after Nov 15, 2023
|Hosted control planes for {product-title} on bare metal
|Technology Preview
|Technology Preview
|Technology Preview
// Needs to move to GA after Nov 15, 2023
|Hosted control planes for {product-title} on {VirtProductName}
|Not Available
|Technology Preview
|Technology Preview

|====

[discrete]
=== Machine management Technology Preview features

.Machine management Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====

> |Feature |4.13 |4.14 |4.15

Author: It's rendering properly for me. Not sure what's up with that.

> |Feature |4.13 |4.14 |4.15

Reviewer: The ones that Joe calls out yesterday LGTM today. I blame Travis.

:_mod-docs-content-type: ASSEMBLY
[id="ocp-4-15-release-notes"]
= {product-title} {product-version} release notes
include::_attributes/common-attributes.adoc[]
:context: release-notes

toc::[]

Red Hat {product-title} provides developers and IT organizations with a hybrid cloud application platform for deploying both new and existing applications on secure, scalable resources with minimal configuration and management overhead. {product-title} supports a wide selection of programming languages and frameworks, such as Java, JavaScript, Python, Ruby, and PHP.

Built on {op-system-base-full} and Kubernetes, {product-title} provides a more secure and scalable multitenant operating system for today's enterprise-class applications, while delivering integrated application runtimes and libraries. {product-title} enables organizations to meet security, privacy, compliance, and governance requirements.

[id="ocp-4-15-about-this-release"]
== About this release

// TODO: Update with the relevant information closer to release.

> {product-title} (link:https://access.redhat.com/errata/RHSA-2023:1326[RHSA-2023:1326]) is now available. This release uses link:https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md[Kubernetes 1.27] with CRI-O runtime. New features, changes, and known issues that pertain to {product-title} {product-version} are included in this topic.

Reviewer 3: ```suggestion
{product-title} (link:https://access.redhat.com/errata/RHSA-2024:xxxx[RHSA-2023:xxxx]) is now available. This release uses link:https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md[Kubernetes 1.27] with CRI-O runtime. New features, changes, and known issues that pertain to {product-title} {product-version} are included in this topic.
```

{product-title} {product-version} clusters are available at https://console.redhat.com/openshift. With the {cluster-manager-first} application for {product-title}, you can deploy {product-title} clusters to either on-premises or cloud environments.

> // Double check OP system versions

Reviewer 3: +1

Author: What does this mean?

[id="ocp-4-15-add-on-support-status"]
== {product-title} layered and dependent component support and compatibility

The scope of support for layered and dependent components of {product-title} changes independently of the {product-title} version. To determine the current support status and compatibility for an add-on, refer to its release notes. For more information, see the link:https://access.redhat.com/support/policy/updates/openshift[Red Hat {product-title} Life Cycle Policy].

[id="ocp-4-15-new-features-and-enhancements"]
== New features and enhancements

This release adds improvements related to the following components and concepts.

[id="ocp-4-15-rhcos"]
=== {op-system-first}

[id="ocp-4-15-installation-and-update"]
=== Installation and update

[id="ocp-4-15-web-console"]
=== Web console

[id="ocp-4-15-openshift-cli"]
=== OpenShift CLI (oc)

> [id="ocp-4-15-ibm-z"]

Reviewer 3: Might be worth putting in a note here about the TM stuff.

=== {ibmzProductName} and {linuxoneProductName}

[id="ocp-4-15-ibm-power"]
=== {ibmpowerRegProductName}

[discrete]
==== {ibmpowerRegProductName} notable enhancements

[discrete]
=== {ibmpowerRegProductName}, {ibmzRegProductName}, and {linuxoneProductName} support matrix

> .{product-title} features

Reviewer 3: Something I thought about was whether or not these tables could be moved to the main docs and then simply referenced here because they take up a lot of space. I don't know though if that works with the user story for these tables.

Author: Probably don't want to do that this time but I'll keep it in mind.

Reviewer: A good iterative step could be to modularize them and add them to the RN assembly. Not suggesting that in this PR (I think it's good to cut the 4.15 RNs just like 4.14 before diverging) but maybe later

|====
|Feature |{ibmpowerProductName} |{ibmzProductName} and {linuxoneProductName}
|Persistent storage using iSCSI
|Supported ^[1]^
|Supported ^[1]^,^[2]^

|Persistent storage using local volumes (LSO)
|Supported ^[1]^
|Supported ^[1]^,^[2]^

|Persistent storage using hostPath
|Supported ^[1]^
|Supported ^[1]^,^[2]^

|Persistent storage using Fibre Channel
|Supported ^[1]^
|Supported ^[1]^,^[2]^

|Persistent storage using Raw Block
|Supported ^[1]^
|Supported ^[1]^,^[2]^

|Persistent storage using EDEV/FBA
|Supported ^[1]^
|Supported ^[1]^,^[2]^
|====
[.small]
--
1. Persistent shared storage must be provisioned by using either {rh-storage-first} or other supported storage protocols.
2. Persistent non-shared storage must be provisioned by using local storage, such as iSCSI, FC, or by using LSO with DASD, FCP, or EDEV/FBA.
--

.Operators
[cols="2,1,1",options="header"]
|====
|Feature |{ibmpowerRegProductName} |{ibmzRegProductName} and {linuxoneProductName}

|Cluster Logging Operator
|Supported
|Supported

|Cluster Resource Override Operator
|Supported
|Supported

|Compliance Operator
|Supported
|Supported

|File Integrity Operator
|Supported
|Supported

> //|HyperShift Operator

Reviewer 3: ```suggestion
|HyperShift Operator
```

Reviewer 3: I think this can all just be added in.

// Note: use [discrete] for these sub-headings.
// example sub-heading below:
//[discrete]
//[id="ocp-4-15-cluster-cloud-controller-manager-operator"]
//=== Cloud controller managers for additional cloud providers

[id="ocp-4-15-deprecated-removed-features"]
== Deprecated and removed features

Some features available in previous releases have been deprecated or removed.

Deprecated functionality is still included in {product-title} and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments. For the most recent list of major functionality deprecated and removed within {product-title} {product-version}, refer to the table below. Additional details for more functionality that has been deprecated and removed are listed after the table.

In the following tables, features are marked with the following statuses:

* _General Availability_
* _Deprecated_
* _Removed_

[discrete]
[id="ocp-4-15-operators-dep-rem"]
=== Operator lifecycle and development deprecated and removed features

.Operator lifecycle and development deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====

> |Feature |4.12 |4.13 |4.14

Reviewer 3: ```suggestion
|Feature |4.13 |4.14 |4.15
```

Reviewer 3: And then all of the tables will need updated in the template to start the process.

|SQLite database format for Operator catalogs
|Deprecated
|Deprecated
|Deprecated

|====

[discrete]
=== Images deprecated and removed features

.Images deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====

> |Feature |4.12 |4.13 |4.14

Reviewer 3: ```suggestion
|Feature |4.13 |4.14 |4.15
```

|`ImageChangesInProgress` condition for Cluster Samples Operator
|Deprecated
|Deprecated
|Deprecated

|`MigrationInProgress` condition for Cluster Samples Operator
|Deprecated
|Deprecated
|Deprecated

|====

//[discrete]
//=== Monitoring deprecated and removed features

//.Monitoring deprecated and removed tracker
//[cols="4,1,1,1",options="header"]
//|====

> //|Feature |4.12 |4.13 |4.14

Reviewer 3: same as above

//|====

[discrete]
=== Installation deprecated and removed features

.Installation deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====

> |Feature |4.12 |4.13 |4.14

Reviewer 3: 4.13 |4.14 |4.15

|Deprecated

|CoreDNS wildcard queries for the `cluster.local` domain
|Deprecated
|Deprecated
|Deprecated

|`compute.platform.openstack.rootVolume.type` for {rh-openstack}
|General Availability
|General Availability
|Deprecated

|`controlPlane.platform.openstack.rootVolume.type` for {rh-openstack}
|General Availability
|General Availability
|Deprecated

|`ingressVIP` and `apiVIP` settings in the `install-config.yaml` file for installer-provisioned infrastructure clusters
|Deprecated
|Deprecated
|Deprecated

|`platform.gcp.licenses` for Google Cloud Provider
|Deprecated
|Deprecated
|Removed

|VMware ESXi 7.0 Update 1 or earlier
|General Availability
|Removed ^[1]^
|Removed

|vSphere 7.0 Update 1 or earlier
|Deprecated
|Removed ^[1]^
|Removed

|====
[.small]
--
1. For {product-title} {product-version}, you must install the {product-title} cluster on a VMware vSphere version 7.0 Update 2 or later instance, including VMware vSphere version 8.0, that meets the requirements for the components that you use.
--
//[discrete]
//=== Updating clusters deprecated and removed features

//.Updating clusters deprecated and removed tracker
//[cols="4,1,1,1",options="header"]
//|====

> //|Feature |4.12 |4.13 |4.14

Reviewer 3: same

//|====

[discrete]
=== Storage deprecated and removed features

.Storage deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====

> |Feature |4.12 |4.13 |4.14

Reviewer 3: same

[discrete]
[id="ocp-4-15-olm-bug-fixes"]
==== Operator Lifecycle Manager (OLM)

[discrete]
[id="ocp-4-15-openshift-api-server-bug-fixes"]
==== OpenShift API server

[discrete]
[id="ocp-4-15-rhcos-bug-fixes"]
==== {op-system-first}

[discrete]
[id="ocp-4-15-scalability-and-performance-bug-fixes"]
==== Scalability and performance

[discrete]
[id="ocp-4-15-storage-bug-fixes"]
==== Storage

[discrete]
[id="ocp-4-15-windows-containers-bug-fixes"]
==== Windows containers
// Added after OCP GA

[id="ocp-4-15-technology-preview"]
== Technology Preview features

Some features in this release are currently in Technology Preview. These experimental features are not intended for production use. Note the following scope of support on the Red Hat Customer Portal for these features:

link:https://access.redhat.com/support/offerings/techpreview[Technology Preview Features Support Scope]

In the following tables, features are marked with the following statuses:

* _Technology Preview_
* _General Availability_
* _Not Available_
* _Deprecated_

[discrete]
=== Networking Technology Preview features

.Networking Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====

> |Feature |4.12 |4.13 |4.14

Reviewer 3: ```suggestion
|Feature |4.13 |4.14 |4.15
```

> |PTP dual NIC hardware configured as boundary clock

Reviewer 3: Anything that is 3 GAs gets removed from the tables. Same for Rem in the dep/rem tables. 3 `Remove` and it is omitted.

|Technology Preview
|General Availability
|General Availability

|Ingress Node Firewall Operator
|Technology Preview
|Technology Preview
|General Availability

|Advertise using L2 mode the MetalLB service from a subset of nodes, using a specific pool of IP addresses
|Technology Preview
|Technology Preview
|Technology Preview

|Multi-network policies for SR-IOV networks
|Technology Preview
|Technology Preview
|Technology Preview

|OVN-Kubernetes network plugin as secondary network
|Not Available
|Technology Preview
|General Availability

|Updating the interface-specific safe sysctls list
|Technology Preview
|Technology Preview
|Technology Preview

> |MT2892 Family [ConnectX-6 Dx] SR-IOV support

Reviewer 3: This would come out too.

|General Availability
|General Availability

|Cron job time zones
|Technology Preview
|Technology Preview
|Technology Preview

|`MaxUnavailableStatefulSet` featureset
|Not Available
|Not Available
|Technology Preview

|====

[discrete]
=== Multi-Architecture Technology Preview features

.Multi-Architecture Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|IBM Secure Execution on {ibmzProductName} and {linuxoneProductName}
|Technology Preview
|General Availability
|General Availability

|{ibmpowerProductName} Virtual Server using installer-provisioned infrastructure
|Not Available
|Technology Preview
|Technology Preview

|`kdump` on `arm64` architecture
|Technology Preview
|Technology Preview
|Technology Preview

|`kdump` on `s390x` architecture
|Technology Preview
|Technology Preview
|Technology Preview

|`kdump` on `ppc64le` architecture
|Technology Preview
|Technology Preview
|Technology Preview

|====

[discrete]
=== Specialized hardware and driver enablement Technology Preview features

.Specialized hardware and driver enablement Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

> |Driver Toolkit

Reviewer 3: This was given an exception but I think we can remove it now.

|Java-based Operator
|Technology Preview
|Technology Preview
|Technology Preview

|====

[discrete]
=== Monitoring Technology Preview features

.Monitoring Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|Alerting rules based on platform monitoring metrics
|Technology Preview
|Technology Preview
|General Availability

|Metrics Collection Profiles
|Not Available
|Technology Preview
|Technology Preview

|====

[discrete]
=== {rh-openstack-first} Technology Preview features

.{rh-openstack} Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|External load balancers with installer-provisioned infrastructure
|Not Available
|Technology Preview
|General Availability

|Dual-stack networking with installer-provisioned infrastructure
|Not Available
|Not Available
|Technology Preview

|====

[discrete]
=== Architecture Technology Preview features

.Architecture Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|Hosted control planes for {product-title} on Amazon Web Services (AWS)
|Technology Preview
|Technology Preview
|Technology Preview

> // Needs to move to GA after Nov 15, 2023

Reviewer 3: You can probably go ahead and update these.

|Technology Preview
|Technology Preview
|Technology Preview

|Cloud controller manager for IBM Cloud Power VS
|Not Available
|Technology Preview
|Technology Preview

|Cloud controller manager for Microsoft Azure
|Technology Preview
|Technology Preview
|General Availability

|Cloud controller manager for Nutanix
|Technology Preview
|General Availability
|General Availability

|Cloud controller manager for VMware vSphere
|Technology Preview
|General Availability
|General Availability

|====

[discrete]
=== Authentication and authorization Technology Preview features

.Authentication and authorization Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|Pod security admission restricted enforcement
|Technology Preview
|Technology Preview
|Technology Preview

|====

[discrete]
=== Machine Config Operator Technology Preview features

.Machine Config Operator Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|{op-system-first} image layering
|Technology Preview
|General Availability
|General Availability

|====

[id="ocp-4-15-known-issues"]
== Known issues

> // TODO: This known issue should carry forward to 4.8 and beyond! This needs some SME/QE review before being updated for 4.11. Need to check if KI should be removed or should stay.

Reviewer 3: This is going to get a KCS for 4.15 so we can remove all this information.

Red Hat Customer Portal users can enable errata notifications in the account settings for Red Hat Subscription Management (RHSM). When errata notifications are enabled, users are notified through email whenever new errata relevant to their registered systems are released.

[NOTE]
====
Red Hat Customer Portal user accounts must have systems registered and consuming {product-title} entitlements for {product-title} errata notification emails to generate.
====

This section will continue to be updated over time to provide notes on enhancements and bug fixes for future asynchronous errata releases of {product-title} {product-version}. Versioned asynchronous releases, for example with the form {product-title} {product-version}.z, will be detailed in subsections. In addition, releases in which the errata text cannot fit in the space provided by the advisory will be detailed in subsections that follow.

[IMPORTANT]
====
For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

//Update with relevant advisory information
[id="ocp-4-15-0-ga"]

> === RHSA-YYYY:XXXX - {product-title} {product-version}.0 image release, bug fix, and security update advisory

Reviewer 3: ```suggestion
=== RHSA-2024:XXXX - {product-title} {product-version}.0 image release, bug fix, and security update advisory
```

Issued: TBD

> {product-title} release {product-version}.0, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-YYYY:XXXX[RHSA-YYYY:XXXX] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-YYYY:XXXX[RHSA-YYYY:XXXX] advisory.

Reviewer 3: ```suggestion
{product-title} release {product-version}.0, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:XXXX[RHSA-2024:XXXX] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:XXXX[RHSA-2024:XXXX] advisory.
```
